app/subscriber/src/features/search-page/components/advanced-search/AdvancedSearch.tsx

```diff
@@ -107,8 +111,17 @@ export const AdvancedSearch: React.FC<IAdvancedSearchProps> = ({ onSearch }) =>
 
   const isAdmin = userInfo?.roles.includes(Claim.administrator);
   const [{ sources, series, mediaTypes }] = useLookup();
+  const [initialState, setInitialState] = React.useState<IFilterSettingsModel | null>(search);
+  const [isFirstLoad, setIsFirstLoad] = React.useState<boolean>(true);
```

Comment:
Included back this logic to handle `save changes` and `saved` labels. The logic compares the previous state of the search terms or advanced filters. If there is a difference, then `save changes` label is shown. On click, the changes are saved and `saved` label is shown.